LoRaEngine/modules/LoRaWanNetworkSrvModule/LoRaWan.NetworkServer/ConcentratorDeduplication.cs

```diff
@@ -105,21 +97,20 @@ private static string CreateCacheKey(LoRaPayloadData payload)
             BinaryPrimitives.WriteUInt32LittleEndian(buffer, BinaryPrimitives.ReadUInt32LittleEndian(payload.DevAddr.Span));
             index += payload.DevAddr.Length;
 
-            if (!payload.Mic.IsEmpty)
-                BinaryPrimitives.WriteUInt32LittleEndian(buffer[index..], BinaryPrimitives.ReadUInt32LittleEndian(payload.Mic.Span));
+            BinaryPrimitives.WriteUInt32LittleEndian(buffer[index..], BinaryPrimitives.ReadUInt32LittleEndian(payload.Mic.Span));
             index += payload.Mic.Length;
 
             payload.RawMessage?.CopyTo(buffer[index..]);
             index += payload.RawMessage?.Length ?? 0;
 
             BinaryPrimitives.WriteUInt16LittleEndian(buffer[index..], BinaryPrimitives.ReadUInt16LittleEndian(payload.Fcnt.Span));
 
-            var key = Sha256.ComputeHash(buffer.ToArray());
+            var key = HashKey(buffer.ToArray());
 
             return BitConverter.ToString(key);
         }
 
-        private static string CreateCacheKey(LoRaPayloadJoinRequest payload)
+        internal string CreateCacheKey(LoRaPayloadJoinRequest payload)
```

Comment:
This can be static.

Reply:
Changed e987b029bcbb7d57399eac92758876fe11d6a2f1